include/deal.II/lac/sparse_direct.h

```diff
@@ -427,6 +432,119 @@ class SparseDirectUMFPACK : public EnableObserverPointer
   std::vector<double> control;
 };
 
+
+/**
+ * This class provides an interface to the parallel sparse direct solver <a
+ * href="https://mumps-solver.org">MUMPS</a>. MUMPS is direct method based on
+ * a multifrontal approach, which performs a direct LU factorization. The
+ * matrix coming in may have either symmetric or nonsymmetric sparsity
+ * pattern.
+ *
+ * @note This class is useable if and only if a working installation of <a
+ * href="https://mumps-solver.org">MUMPS</a> exists on your system and was
+ * detected during configuration of <code>deal.II</code>.
+ *
+ * <h4>Instantiations</h4>
+ *
+ * There are instantiations of this class for SparseMatrix<double>,
+ * SparseMatrix<float>, BlockSparseMatrix<double>, and
+ * BlockSparseMatrix<float>.
+ */
+class SparseDirectMUMPS : public EnableObserverPointer
+{
+private:
+#ifdef DEAL_II_WITH_MUMPS
+  DMUMPS_STRUC_C id;
+#endif // DEAL_II_WITH_MUMPS
+
+  double                 *a;
+  std::vector<double>     rhs;
+  int                    *irn;
+  int                    *jcn;
+  types::global_dof_index n;
+  types::global_dof_index nz;
+
+  /**
+   * This function initializes a MUMPS instance and hands over the system's
+   * matrix <tt>matrix</tt>.
+   */
+  template <class Matrix>
+  void
+  initialize_matrix(const Matrix &matrix);
+
+  /**
+   * Copy the computed solution into the solution vector.
+   */
+  void
+  copy_solution(Vector<double> &vector);
+
+  /**
+   *
+   */
+  void
+  copy_rhs_to_mumps(const Vector<double> &rhs);
+
+  /**
+   * Flags storing whether the function <tt>initialize ()</tt> has already
+   * been called.
+   */
+  bool initialize_called;
+
+public:
+  /**
+   * Declare type for container size.
+   */
+  using size_type = types::global_dof_index;
+
+  /**
+   * Constructor
+   */
+  SparseDirectMUMPS();
+
+  /**
+   * Destructor
+   */
+  ~SparseDirectMUMPS();
+
+  /**
+   * Exception
+   */
+  DeclException0(ExcInitializeAlreadyCalled);
+
+  /**
+   * This function initializes a MUMPS instance and hands over the system's
+   * matrix <tt>matrix</tt> and right-hand side <tt>rhs_vector</tt> to the
+   * solver.
+   */
+  template <class Matrix>
+  void
+  initialize(const Matrix &matrix, const Vector<double> &rhs_vector);
+
+  /**
+   * This function initializes a MUMPS instance and computes the factorization
+   * of the system's matrix <tt>matrix</tt>.
+   */
+  template <class Matrix>
+  void
+  initialize(const Matrix &matrix);
+
+  /**
+   * A function in which the linear system is solved and the solution vector
+   * is copied into the given <tt>vector</tt>. The right-hand side need to be
+   * supplied in initialize(matrix, vector);
+   */
+  void
+  solve(Vector<double> &vector);
+
+  /**
+   * A function in which the inverse of the matrix is applied to the input
+   * vector <tt>src</tt> and the solution is written into the output vector
+   * <tt>dst</tt>.
+   */
+  void
+  vmult(Vector<double> &dst, const Vector<double> &src);
```

Comment:
A next step would be to extend this interface to generic vector types, in particular deal.II's `LinearAlgebra::distributed::Vector`.